### ISSUE TITLE
W0212 False positive when using self.__class__

### ISSUE THREAD

**Issue by markastern** [PyCQA/pylint#11160]
### Bug description

```python
class C:
    _X = None

    def f(self):
        if self.__class__._X is None:
            pass

self.__class__._X is clearly in the same class, so it should not complain. Instead, I get W0212.
```

### Configuration

```ini

```

### Command used

```shell
pylint /tmp/bug.py
```

### Pylint output

```python
************* Module bug
/tmp/bug.py:1:0: C0114: Missing module docstring (missing-module-docstring)
/tmp/bug.py:1:0: C0115: Missing class docstring (missing-class-docstring)
/tmp/bug.py:4:4: C0116: Missing function or method docstring (missing-function-docstring)
/tmp/bug.py:5:11: W0212: Access to a protected member _X of a client class (protected-access)
/tmp/bug.py:1:0: R0903: Too few public methods (1/2) (too-few-public-methods)

------------------------------------------------------------------
Your code has been rated at 0.00/10 (previous run: 0.00/10, +0.00)
```

### Expected behavior

As above, but without:
/tmp/bug.py:5:11: W0212: Access to a protected member _X of a client class (protected-access)

### Pylint version

```shell
pylint 3.3.9
astroid 3.3.8
Python 3.14.5 (main, May 11 2026, 00:00:00) [GCC 15.2.1 20260123 (Red Hat 15.2.1-7)]
```

### OS / Environment

Fedora 43

### Additional dependencies

```python

```

**Comment by Hibou57** [comment]
It behave differently with class-method:

    class C:
        _X = None

        @classmethod
        def f(cls):
            if cls._X is None:
                pass

PyLint sees nothing wrong
